sycl/include/sycl/ext/intel/esimd/memory.hpp

```diff
@@ -4371,8 +4371,9 @@ __ESIMD_API std::enable_if_t<
     __ESIMD_DNS::get_num_args<Op>() == 1 &&
         ext::oneapi::experimental::is_property_list_v<PropertyListT>,
     simd<T, N>>
-atomic_update(T *p, simd<Toffset, N> byte_offset, simd_view<T, RegionTy> src0,
-              simd_mask<N> mask, PropertyListT props = {}) {
+atomic_update(T *p, simd<Toffset, N> byte_offset,
+              simd_view<simd<T, N>, RegionTy> src0, simd_mask<N> mask,
```

Comment:
Why this is changed? It is not related to the accessors. This kind of changes may be done separately.

Also, this argument definition is not correct. The first argument is the type of viewed object.
It may be simd<T, N> if it is the result of select<N,1>. But it also may be simd<T,N*10> if it is the result of select<N,10>(). Also, it may even have a different type `T` if `simd_view` is the result of bit_cast_view<T>().

Reply:
I agree that the original/current source is also incorrect.
Let's fix those kind of errors in a separate PR, not mix read-write checks for accessors and simd_view overloads.